### PR TITLE
[jquery/v2] Add types to and deprecate key props

### PR DESCRIPTION
**Pull request by oBusk**
I found it somewhat annoying that JQuery extends some default event properties but not having them `@deprecated` as the original is in [`lib.dom.d.ts`](https://github.com/Microsoft/TypeScript/blob/master/lib/lib.dom.d.ts)

Firstly, `char` and `key` are typed `any`, they should be typed as `string` as they are in `lib.dom.d.ts`
* `char`- https://github.com/Microsoft/TypeScript/blob/v3.2.2/lib/lib.dom.d.ts#L9304
* `key` - https://github.com/Microsoft/TypeScript/blob/v3.2.2/lib/lib.dom.d.ts#L9309

Secondly [`which`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/which), [`keyCode`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keycode), `char` and [`charCode`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/charCode) are nominally deprecated but still widely supported.

Most interestingly, `char`, `charCode` and `keyCode` are `@depracted` in `lib.dom.d.ts`.
* `char` - https://github.com/Microsoft/TypeScript/blob/v3.2.2/lib/lib.dom.d.ts#L9303-L9304
* `charCode` - https://github.com/Microsoft/TypeScript/blob/v3.2.2/lib/lib.dom.d.ts#L9305-L9306
* `keyCode` - https://github.com/Microsoft/TypeScript/blob/v3.2.2/lib/lib.dom.d.ts#L9310-L9311

And since these properties are not modified or polyfilled in jquery
* https://github.com/jquery/jquery/blob/2.2.4/src/event.js#L402

It would make sense if they are marked `@deprecated` in jQuery aswell, matching using _raw_ javascript events.

Furthermore, `which` _is_ modified and "polyfilled" by jQuery but it relies on `keyCode` _or_ `charCode` 
* https://github.com/jquery/jquery/blob/2.2.4/src/event.js#L406-L408

Should also be marked `@deprecated` as it relies on `@deprecated` properties.

The idea of this is to match `lib.dom.d.ts` better by softly deprecating the properties, it shouldn't break any builds but at potentially add some linting warnings depending on peoples lint setup.

### CODE REVIEW

**Comment by oBusk** [comment]
All these errors in jquery types should probably not be fixed as part of this PR. Can't see any PR to fix them elsewhere? Might do it as a separate PR.

**Comment by oBusk** [comment]
Created a PR #31527 to fix the lint errors so that this PR can have a better diff.

File: types/jquery/legacy.d.ts [hunk 1]
```diff
@@ -167,9 +167,9 @@ interface JQueryMouseEventObject extends JQueryInputEventObject {
  * @deprecated ​ Deprecated. Use \`{@link JQuery.Event }\`.
  */
 interface JQueryKeyEventObject extends JQueryInputEventObject {
-    char: any;
+    char: string;
     charCode: number;
-    key: any;
+    key: string;
     keyCode: number;
```
Comment (leonard-thieu) [inline types/jquery/legacy.d.ts]: Even though the interface itself is deprecated, it's probably still worth it to deprecate the individual properties too. I imagine there could be scenarios where tooling may not alert that a deprecated type is being used but would on a deprecated property.

Also, the deprecations serve different purposes. The one on the type encourages users to use the more accurate `JQuery.Event` type and only has compile-time consequences. The ones on the properties are to notify about potential loss of support and has run-time consequences.

**Comment by oBusk** [comment]
@leonard-thieu I also added the type and deprecation message for `jquery/v1`! 

Should there be any comment on the deprecated properties? Like referring to use `key` or documentation?

**Comment by leonard-thieu** [comment]
I'm not sure on what the proper guidance is. If you're confident in the guidance that you'd supply, then I say go for it, otherwise, leave it empty to match `lib.dom.d.ts`.

**Pull request merged**